pkg/bucketclass/reconciler.go

```diff
@@ -297,3 +298,76 @@ func (r *Reconciler) FinalizeDeletion() error {
 	}
 	return nil
 }
+
+// UpdateBucketClass updates all buckets that are assigned to a BucketClass
+func (r *Reconciler) UpdateBucketClass() error {
+	log := r.Logger
+
+	if r.BucketClass == nil {
+		return fmt.Errorf("BucketClass not loaded %#v", r)
+	}
+
+	policyTiers := []nb.TierItem{}
+	tiers := []nb.TierInfo{}
+
+	for i := range r.BucketClass.Spec.PlacementPolicy.Tiers {
+		tier := &r.BucketClass.Spec.PlacementPolicy.Tiers[i]
+		// Tier is irrelevant and will be populated in the BE
+		policyTiers = append(policyTiers, nb.TierItem{Order: int64(i), Tier: "TEMP"})
+		// we assume either mirror or spread but no mix and the bucket class controller rejects mixed classes.
+		placement := "SPREAD"
+		if tier.Placement == nbv1.TierPlacementMirror {
+			placement = "MIRROR"
+		}
+		// Name is irrelevant and will be populated in the BE
+		tiers = append(tiers, nb.TierInfo{Name: "TEMP", AttachedPools: tier.BackingStores, DataPlacement: placement})
+	}
+
+	err := r.NBClient.UpdateBucketClass(nb.UpdateBucketClassParams{
```

Comment:
I suggest that this call will return back the applied bucket class by the core, so that the operator will update the CR if needed.